File: Data Structures & Algorithms/validate-parentheses/submission-3.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isValid(s) {
        const pairs = new Map([
            [")", "("],
            ["]", "["],
            ["}", "{"]
        ]);

        let stack = []

        for (let i = 0; i < s.length; ++i) {
            if (s[i] === "(" || s[i] === "[" || s[i] === "{") {
                stack.push(s[i])
            }
            else {
                if (stack.length === 0) return false

                if(pairs.get(s[i]) === stack.at(-1)) {
                    stack.pop()
                }
                else return false
            }
        }

        return stack.length === 0
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-3.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        operators = ["/", "-", "+", "*"]

        for token in tokens:
            if token not in operators:
                stack.append(int(token))
                continue

            b = stack.pop()
            a = stack.pop()

            if token == "/":
                stack.append(int(a / b))
            elif token == "+":
                stack.append(a + b)
            elif token == "-":
                stack.append(a - b)
            else:
                stack.append(a * b)

        return stack[-1]